packages/material-ui/src/AppBar/AppBar.js

```diff
@@ -29,71 +58,88 @@ export const styles = (theme) => {
         // Prevent the app bar to be visible on each printed page.
         position: 'absolute',
       },
-    },
+    }),
     /* Styles applied to the root element if `position="absolute"`. */
-    positionAbsolute: {
+    ...(styleProps.position === 'absolute' && {
       position: 'absolute',
       zIndex: theme.zIndex.appBar,
       top: 0,
       left: 'auto',
       right: 0,
-    },
+    }),
     /* Styles applied to the root element if `position="sticky"`. */
-    positionSticky: {
+    ...(styleProps.position === 'sticky' && {
       // ⚠️ sticky is not supported by IE11.
       position: 'sticky',
       zIndex: theme.zIndex.appBar,
       top: 0,
       left: 'auto',
       right: 0,
-    },
+    }),
     /* Styles applied to the root element if `position="static"`. */
-    positionStatic: {
+    ...(styleProps.position === 'static' && {
       position: 'static',
-    },
+    }),
     /* Styles applied to the root element if `position="relative"`. */
-    positionRelative: {
+    ...(styleProps.position === 'relative' && {
       position: 'relative',
-    },
+    }),
     /* Styles applied to the root element if `color="default"`. */
-    colorDefault: {
+    ...(styleProps.color === 'default' && {
       backgroundColor: backgroundColorDefault,
       color: theme.palette.getContrastText(backgroundColorDefault),
-    },
+    }),
     /* Styles applied to the root element if `color="primary"`. */
-    colorPrimary: {
+    ...(styleProps.color === 'primary' && {
       backgroundColor: theme.palette.primary.main,
       color: theme.palette.primary.contrastText,
-    },
+    }),
     /* Styles applied to the root element if `color="secondary"`. */
-    colorSecondary: {
+    ...(styleProps.color === 'secondary' && {
       backgroundColor: theme.palette.secondary.main,
       color: theme.palette.secondary.contrastText,
-    },
+    }),
     /* Styles applied to the root element if `color="inherit"`. */
-    colorInherit: {
+    ...(styleProps.color === 'inherit' && {
       color: 'inherit',
-    },
+    }),
     /* Styles applied to the root element if `color="transparent"`. */
-    colorTransparent: {
+    ...(styleProps.color === 'transparent' && {
       backgroundColor: 'transparent',
       color: 'inherit',
-    },
+    }),
   };
-};
+});
 
-const AppBar = React.forwardRef(function AppBar(props, ref) {
-  const { classes, className, color = 'primary', position = 'fixed', ...other } = props;
+const AppBar = React.forwardRef(function AppBar(inProps, ref) {
+  const {
+    className,
+    color = 'primary',
+    position = 'fixed',
+    classes: inClasses,
+    ...other
+  } = useThemeProps({
+    props: inProps,
+    name: 'MuiAppBar',
+  });
```

Comment:
```suggestion
  const props = useThemeProps({ props: inProps, name: 'MuiAppBar' });
  const { className, color = 'primary', position = 'fixed', ...other } = props;
```

Reply:
I didn't format like that yarn prettier did.

Reply:
Oh missed that I did create props, got it also.